ramorris3/alextown
# Editor

**I MADE A LEVEL EDITOR, SUCKAS**
- It's a WIP, but I have enough functionality for a commit.
- Hopefully this will speed things up for us once we're done with the mechanics and we just need to fill out the game with content.

**TEST PLAN:**
1. Pull the branch (obvs)
2. From the root directory (alextown/) run the server
3. In your browser go to root directory /editor (ex. "localhost:3000/editor")
4. Click around to place chompers.  Make sure you can't place chompers on top of each other, and make sure you can't place them offscreen.
5. Ctrl + S (not command) to "save."
   a. Make sure when you type the file name, all non-alphanumeric characters (except underscore) are stripped from the filename (you'll see the filename after you click "OK").
   b. Make sure that when you save multiple times, you are only prompted for the filename on the first save
   c. Make sure that the console output after the save matches the grid, where 0's are empty spaces, and Z's are chompers.

**Note:**  The grid is not going to match Christian's grid exactly, but it's a very similar idea.  Maybe when we finish building out the editor, we can standardize grid syntax using the editor, and the tweak the way the grid string is parsed to make it match the level editor syntax.  Should be really close to what C already has, though, so that would be a minor tweak.

And then code review of course.

**IDEAS FOR FUTURE FEATURES**
- Saving actually works (nodejs and JQuery AJAX or AngularJS??)
- Left and Right arrows to click that will move the camera 100% to the left or the right, so you can build out a full stage
- A small HUD with the current creature, and left and right arrows for choosing a different creature to drop
- Loading from a file functionality
- "Save As" functionality, and a "Save" and "Save As" button.
- Background art instead of black space?  Maybe.
